<!-- eslint-disable --> does not work in the v1.0.0-beta.6

1\. I've found an error on the v1.0.0-beta.4: `<!-- eslint-disable some-rule -->` completely disables parsing the whole fragment and even the next fragment:

`test.md`
```md
### Heading

Text

` ``js
console.log('a')

` ``

` ``js
console.log('b')

` ``

` ``js
console.log('c')

` ``
```
Error log:
```console
   6:1   error  Use the global form of 'use strict'     strict
   6:17  error  Missing semicolon                       semi
  11:1   error  Use the global form of 'use strict'     strict
  11:17  error  Missing semicolon                       semi
  16:1   error  Use the global form of 'use strict'     strict
  16:17  error  Missing semicolon                       semi
```

`test.md` with comments:
```md

### Heading

Text

<!-- eslint-disable strict -->
` ``js
console.log('a')

` ``

` ``js
console.log('b')

` ``

` ``js
console.log('c')

` ``
```
Error log (`semi` errors for the fragments 1 and 2 are not reported; fragment 2 is ignored):
```
  17:1   error  Use the global form of 'use strict'     strict
  17:17  error  Missing semicolon                       semi
```

2\. So I've tried to update to v1.0.0-beta.6 and make a PR for Node.js. v1.0.0-beta.6 seems to fix this issue, but it has a new breaking one: code blocks with `<!-- eslint-disable -->` are parsed and throws if parsing errors found:
`test.md`:
```md
### Heading

Text

<!-- eslint-disable -->
` ``js
console.log('a') bang

` ``
```
Error log:
```console
  7:18  error  Parsing error: Unexpected token bang
```

--- a/README.md
+++ b/README.md
@@ -102,6 +102,26 @@ Each code block in a file is linted separately, so configuration comments apply 
     alert("Hello, world!");
     ```
 
+## Skipping Blocks
+
+Sometimes it can be useful to have code blocks marked with `js` even though they don't contain valid JavaScript syntax, such as commented JSON blobs that need `js` syntax highlighting. Standard `eslint-disable` comments only silence rule reporting, but ESLint still reports any syntax errors it finds. In cases where a code block should not even be parsed, insert a non-standard `<!-- eslint-skip -->` comment before the block, and this plugin will hide the following block from ESLint. Neither rule nor syntax errors will be reported.
+
+    There are comments in this JSON, so we use `js` syntax for better
+    highlighting. Skip the block to prevent warnings about invalid syntax.
+
+    <!-- eslint-skip -->
+
+    ```js
+    {
+        // This code block is hidden from ESLint.
+        "hello": "world"
+    }
+    ```
+
+    ```js
+    console.log("This code block is linted normally.");
+    ```
+
 ## Unsatisfiable Rules
 
 Since code blocks are not files themselves but embedded inside a Markdown document, some rules do not apply to Markdown code blocks, and messages from these rules are automatically suppressed:

--- a/lib/processor.js
+++ b/lib/processor.js
@@ -62,7 +62,7 @@ function getComment(html) {
         return "";
     }
 
-    return "/*" + html + "*/";
+    return html;
 }
 
 /**
@@ -88,7 +88,12 @@ function preprocess(text) {
                     if (!comment) {
                         break;
                     }
-                    comments.unshift(comment);
+
+                    if (comment.trim() === "eslint-skip") {
+                        return;
+                    }
+
+                    comments.unshift("/*" + comment + "*/");
                     index--;
                     previousNode = parent.children[index];
                 }

--- a/tests/lib/processor.js
+++ b/tests/lib/processor.js
@@ -405,6 +405,61 @@ describe("processor", function() {
             ].join("\n"));
         });
 
+        describe("eslint-skip", function() {
+
+            it("should skip the next block", function() {
+                var code = [
+                    "<!-- eslint-skip -->",
+                    "",
+                    "```js",
+                    "alert('Hello, world!');",
+                    "```"
+                ].join("\n");
+                var blocks = processor.preprocess(code);
+
+                assert.equal(blocks.length, 0);
+            });
+
+            it("should skip only one block", function() {
+                var code = [
+                    "<!-- eslint-skip -->",
+                    "",
+                    "```js",
+                    "alert('Hello, world!');",
+                    "```",
+                    "",
+                    "```js",
+                    "var answer = 6 * 7;",
+                    "```"
+                ].join("\n");
+                var blocks = processor.preprocess(code);
+
+                assert.equal(blocks.length, 1);
+                assert.equal(blocks[0], "var answer = 6 * 7;");
+            });
+
+            it("should still work surrounded by other comments", function() {
+                var code = [
+                    "<!-- eslint-disable no-console -->",
+                    "<!-- eslint-skip -->",
+                    "<!-- eslint-disable quotes -->",
+                    "",
+                    "```js",
+                    "alert('Hello, world!');",
+                    "```",
+                    "",
+                    "```js",
+                    "var answer = 6 * 7;",
+                    "```"
+                ].join("\n");
+                var blocks = processor.preprocess(code);
+
+                assert.equal(blocks.length, 1);
+                assert.equal(blocks[0], "var answer = 6 * 7;");
+            });
+
+        });
+
     });
 
     describe("postprocess", function() {

